### ISSUE TITLE
Http IIS 502.5 Error after migrating from ASP.NET Core 1.0 to 1.1

### ISSUE THREAD

**Issue by saf-itpro** [dotnet/efcore#7058]
### Steps to reproduce
I used these [official ASP.NET Articles](https://blogs.msdn.microsoft.com/dotnet/2016/11/16/announcing-net-core-1-1/) [and](https://blogs.msdn.microsoft.com/dotnet/2016/11/16/announcing-entity-framework-core-1-1/) to upgrade my ASP.NET Core 1.0 app to 1.1. The app compiles fine but when I run the app using F5, I get the following error

### The issue
```HTTP Error 502.5 - Process Failure

Common causes of this issue:

The application process failed to start
The application process started but then stopped
The application process started but failed to listen on the configured port

```
**NOTE**:
My app is on development machine(windows 10)  running on default built-in IIS express. No server or migration to a different sever is involved.

**Windows Event Log**:

    Failed to start process with commandline '"C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\Extensions\Microsoft\Web Tools\ProjectSystem\VSIISExeLauncher.exe" -p "C:\Program Files\dotnet\dotnet.exe" -a "C:\DotNet2015\src\ABCTest\bin\Debug\netcoreapp1.1\ABCTest.dll" -wd "C:\DotNet2015\src\ABCTest" -wbe Global\Build30603dd4-5933-4547-8aff-af488d42458c -lse Global\Launchf9d250ad-37e9-4e5b-8375-46f63c747deb', ErrorCode = '0x80004005'.

### Further technical details

EF Core version: 1.1
Operating system: Windows 10
Visual Studio version: 2015-Update 3 (with latest updates)

Other details about my project setup:

**project.json File**:
{
  "userSecretsId": "aspnet-ABCTest-6af8ade3-87ff-4468-a9ce-8bb69c696ab8",

  "dependencies": {
    "Microsoft.NETCore.App": {
      "version": "1.1.0",
      "type": "platform"
    },
    "Microsoft.EntityFrameworkCore.SqlServer": "1.1.0",
    "Microsoft.EntityFrameworkCore.Tools": "1.1.0-preview4-final",
    "Microsoft.AspNetCore.Mvc": "1.1.0",
    "Microsoft.AspNetCore.Mvc.Core": "1.1.0",
    "Microsoft.AspNetCore.Server.Kestrel": "1.1.0",
    "Microsoft.EntityFrameworkCore.SqlServer.Design": "1.1.0",
    "Microsoft.AspNetCore.Authentication.Cookies": "1.1.0",
    "Microsoft.AspNetCore.Diagnostics": "1.1.0",
    "Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore": "1.1.0",
    "Microsoft.AspNetCore.Identity": "1.1.0",
    "Microsoft.AspNetCore.Identity.EntityFrameworkCore": "1.1.0",
    "Microsoft.AspNetCore.Razor.Tools": "1.1.0-preview4-final",
    "Microsoft.AspNetCore.Server.IISIntegration": "1.1.0",
    "Microsoft.AspNetCore.Server.IISIntegration.Tools": "1.1.0-preview4-final",
    "Microsoft.AspNetCore.StaticFiles": "1.1.0",
    "Microsoft.Extensions.Configuration.EnvironmentVariables": "1.1.0",
    "Microsoft.Extensions.Configuration.Json": "1.1.0",
    "Microsoft.Extensions.Configuration.UserSecrets": "1.1.0",
    "Microsoft.Extensions.Logging": "1.1.0",
    "Microsoft.Extensions.Logging.Console": "1.1.0",
    "Microsoft.Extensions.Logging.Debug": "1.1.0",
    "Microsoft.Extensions.Options.ConfigurationExtensions": "1.1.0",
    "Microsoft.VisualStudio.Web.BrowserLink.Loader": "14.1.0",
    "Microsoft.VisualStudio.Web.CodeGeneration.Tools": "1.1.0-preview4-final",
    "Microsoft.VisualStudio.Web.CodeGenerators.Mvc": "1.1.0-preview4-final"  },

  "tools": {
    "BundlerMinifier.Core": "2.0.238"  },

  "frameworks": {
    "netcoreapp1.1": {
      "imports": [
        "dotnet5.6",
        "portable-net45+win8"
      ]
    }
  },

  "buildOptions": {
    "emitEntryPoint": true,
    "preserveCompilationContext": true
  },

  "runtimeOptions": {
    "configProperties": {
      "System.GC.Server": true
    }
  },

  "publishOptions": {
    "include": [
      "wwwroot",
      "Views",
      "Areas/**/Views",
      "appsettings.json",
      "web.config"
    ]
  },

  "scripts": {
    "prepublish": [ "bower install", "dotnet bundle" ],
    "postpublish": [ "dotnet publish-iis --publish-folder %publish:OutputPath% --framework %publish:FullTargetFramework%" ]
  }
}

**Comment by AndriySvyryd** [comment]
Try deleting all the files in the publish folder before running it again. If publishing to Azure use a new slot.

**Comment by halter73** [comment]
I would also try to [enable stdout/stderror logging in your web.config](https://docs.microsoft.com/en-us/aspnet/core/hosting/aspnet-core-module#log-creation-and-redirection). That output should show you what exception is causing the process to crash.

You can probably get the same info by navigating to your project directory in a shell and doing `dotnet run`.

**Comment by saf-itpro** [comment]
@AndriySvyryd My app is on development machine running on Visual Studio 2015 with built-in IIS Express on Windows 10 . There is no Publish folder since I've not yet published it to anywhere.

**Comment by AndriySvyryd** [comment]
@saf-itpro In your case the app runs from the bin folder under your project, so delete it to make sure everything gets updated

**Comment by saf-itpro** [comment]
@AndriySvyryd I deleted the `bin` folder ran the `clean solution`, compiled the app and ran again. Still the same error.

**Comment by AndriySvyryd** [comment]
@saf-itpro Then please post the log as @halter73 suggested

**Comment by saf-itpro** [comment]
@AndriySvyryd I ran dotnet run command from the src directory and got the following error:
[Project ABCTest (.NETCoreApp,Version=v1.1) was previously compiled. Skipping compilation.
The specified framework 'Microsoft.NETCore.App', version '1.1.0' was not found.
  - Check application dependencies and target a framework version installed at:
      C:\Program Files\dotnet\shared\Microsoft.NETCore.App
  - The following versions are installed:
      1.0.0
      1.0.1
  - Alternatively, install the framework version '1.1.0'.](url)

**Comment by AndriySvyryd** [comment]
@saf-itpro You need to install .NET Core 1.1 SDK https://go.microsoft.com/fwlink/?LinkID=835014

**Comment by saf-itpro** [comment]
@AndriySvyryd Let me do that. I thought when I got a notification in VS about the release of 1.1, it had downloaded an exe that I had already ran; and hence that was enough along with the instructions in the two above mentioned articles.

**Comment by AndriySvyryd** [comment]
@saf-itpro Unfortunately the current VS Tools only include the .NET Core 1.0.1 SDK

**Comment by saf-itpro** [comment]
@AndriySvyryd Thank you for coming to my rescue. It's working now after I installed .NET Core 1.1 SDK that you sent the link of. Your team may want to update the articles I mentioned above with the the instructions of installing .NET Core 1.1 SDK as a pre-requisite.

**Comment by jasoncoe2** [comment]
I have hit a very similar issue with ASP.net Core 2.0.  I had copied my VS project to a new one, and i was getting this error message.  After doing some research, I was able to determine that the nlog.config file was not copied into the bin > Debug > net461 folder.  Once I did this, I was able to run my application.  I found it by running dotnet run from the command line on my project where the csproj files live.  I hope it helps any future people with this issue.
